.travis.yml

```diff
@@ -2,82 +2,61 @@ language: python
 
 matrix:
   include:
-  - python: 3.5
-    env: export FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='False'
-  - python: 3.5
-    env: export FAIL_ON_EXTERNAL_DEPRECATION='True'; MINIMAL_ENV='False'
-  - python: 3.5
-    env: export FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='True'
-  - os: osx
+  - env: export PYTHON=3.5; FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='False'
+  - env: export PYTHON=3.5; FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='False'; MPLBACKEND="agg"
+    os: osx
+    language: generic
+  - env: export PYTHON=3.6; FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='False'
+  - env: export PYTHON=3.6; FAIL_ON_EXTERNAL_DEPRECATION='True'; MINIMAL_ENV='False'
+  - env: export PYTHON=3.6; FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='True'
+  - env: export PYTHON=3.6; FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='False'; MPLBACKEND="agg"
+    os: osx
     language: generic
-    python: 3.5
-    env: export FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='False'
   allow_failures:
-  - env: export FAIL_ON_EXTERNAL_DEPRECATION='True'; MINIMAL_ENV='False'
+  - env: export PYTHON=3.6; FAIL_ON_EXTERNAL_DEPRECATION='True'; MINIMAL_ENV='False'
 
 before_install:
   - if [[ $TRAVIS_OS_NAME == 'osx' ]] ; then
       curl "https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh" -o miniconda.sh;
     else wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
     fi
-  - if [[ $MINIMAL_ENV == 'False' ]] ; then
-      chmod +x miniconda.sh;
-      ./miniconda.sh -b -p $HOME/miniconda;
-      export "PATH=$HOME/miniconda/bin:$PATH";
-      hash -r;
-      conda update --yes conda;
-    fi
+  - chmod +x miniconda.sh;
+    ./miniconda.sh -b -p $HOME/miniconda;
+    export "PATH=$HOME/miniconda/bin:$PATH";
+    hash -r;
 install:
 
   - if [[ $MINIMAL_ENV == 'False' ]] ; then
-      DEPS="pytest pytest-cov numpy scipy matplotlib ipython h5py sympy scikit-learn dill natsort setuptools scikit-image cython lxml ipyparallel dask";
-      conda create -n testenv --yes $DEPS;
-      source activate testenv;
-      conda install pip;
-      if [[ $TRAVIS_OS_NAME == 'linux' ]]; then
-        pip uninstall -y matplotlib;
-        pip install matplotlib;
-      fi;
-      python setup.py install;
-      pip install pytest pytest-xdist;
-    else pip install pytest;
-    pip3 install .;
+      DEPS="pip numpy scipy matplotlib ipython h5py sympy scikit-learn dill natsort setuptools scikit-image cython lxml ipyparallel dask traits traitsui";
+    else DEPS="pip ipython numpy scipy matplotlib h5py sympy scikit-image";
```

Comment:
this make no sense, why we need to install `ipython numpy scipy matplotlib h5py` in the minimal environment?
We should not do that instead let to pull the things which are declared in setup.py.
Else we are only pretending that we test it on minimal environment.